pkg/cmd/provisioning/azure/create_oidc_issuer_test.go

```diff
@@ -0,0 +1,542 @@
+				}
+				mockUpdateStorageAccountSuccess(wrapper, testOIDCResourceGroupName, testStorageAccountName, testRegionName, testSubscriptionID, expectedTags)
+				return wrapper
+			},
+			expectError: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+
+			mockAzureClientWrapper := test.mockAzureClientWrapper(mockCtrl)
+			resourceTags := map[string]string{
+				fmt.Sprintf("%s_%s", ownedAzureResourceTagKeyPrefix, testInfraName): ownedAzureResourceTagValue,
+				"testtagname0": "testtagvalue0",
+				"testtagname1": "testtagvalue1",
+			}
+			storageAccount, err := ensureStorageAccount(mockAzureClientWrapper, testStorageAccountName, testOIDCResourceGroupName, testRegionName, resourceTags)
+			if test.expectError {
+				require.Error(t, err, "expected error")
+			} else {
+				require.NoError(t, err, "unexpected error")
+				require.NotNil(t, storageAccount, "expected storageAccount to not be nil")
+			}
+		})
+	}
+}
+
+func mockAzureClientWrapper(mockCtrl *gomock.Controller) *azureclients.AzureClientWrapper {
+	wrapper := azureclients.AzureClientWrapper{}
+	wrapper.ResourceGroupsClient = mockazure.NewMockResourceGroupsClient(mockCtrl)
+	wrapper.StorageAccountClient = mockazure.NewMockAccountsClient(mockCtrl)
+	wrapper.BlobContainerClient = mockazure.NewMockBlobContainersClient(mockCtrl)
+	// BlobSharedKeyClient is not set by azureclients.NewAzureClientWrapper because we won't
+	// have storage account keys needed to instantiate the client. When BlobSharedKeyClient is previously
+	// set as it is here, uploadOIDCDocuments will use wrapper.BlobSharedKeyClient and will not create
+	// a real client.
+	wrapper.BlobSharedKeyClient = mockazure.NewMockAZBlobClient(mockCtrl)
+	wrapper.UserAssignedIdentitiesClient = mockazure.NewMockUserAssignedIdentitiesClient(mockCtrl)
+	wrapper.RoleDefinitionsClient = mockazure.NewMockRoleDefinitionsClient(mockCtrl)
+	wrapper.RoleAssignmentClient = mockazure.NewMockRoleAssignmentsClient(mockCtrl)
+	wrapper.FederatedIdentityCredentialsClient = mockazure.NewMockFederatedIdentityCredentialsClient(mockCtrl)
+	// Mock = true so that runtime.Poller operations will be mocked by an azureclients.PollerWrapper
+	wrapper.Mock = true
+	return &wrapper
+}
+
+func mockAzureClientWrapperWithStorageClientBeginCrateResp(mockCtrl *gomock.Controller, mockStorageClientBeginCreateResp *armstorage.AccountsClientCreateResponse) *azureclients.AzureClientWrapper {
+	wrapper := mockAzureClientWrapper(mockCtrl)
+	if mockStorageClientBeginCreateResp != nil {
+		wrapper.MockStorageClientBeginCreateResp = *mockStorageClientBeginCreateResp
+	}
+	return wrapper
+}
+
+func mockCreateOrUpdateResourceGroupSuccess(wrapper *azureclients.AzureClientWrapper, resourceGroupName, subscriptionID string) {
+	wrapper.ResourceGroupsClient.(*mockazure.MockResourceGroupsClient).EXPECT().CreateOrUpdate(gomock.Any(), resourceGroupName, gomock.Any(), gomock.Any()).Return(
```

Comment:
Code review: write matcher such that we can check the actual tags being passed in as parameters. The tags are stored within a `map[string]*string` and without something custom we couldn't test the values of the tags as pointers to addresses.